crates/hash/src/multi_digest.rs

```diff
@@ -172,6 +173,60 @@ impl<D: Digest + Send + Sync + Clone> ParallelDigest for D {
 	}
 }
 
+#[derive(Clone)]
+pub struct MultipleDigests<D: Digest, const N: usize>([D; N]);
+
+impl<D: Digest + Send + Sync + Clone + digest::Reset, const N: usize> MultiDigest<N>
+	for MultipleDigests<D, N>
+{
+	type Digest = D;
+
+	fn new() -> Self {
+		Self(array::from_fn(|_| D::new()))
+	}
+
+	fn update(&mut self, data: [&[u8]; N]) {
+		self.0.par_iter_mut().enumerate().for_each(|(i, hasher)| {
```

Comment:
I personally would not use parallel iterator inside this implementation, because "normal" implementations of `MultiDigest` are single-threaded and we have a level of parallelization somewhere on other levels of the call-stack